pilot/pkg/xds/delta.go

```diff
@@ -456,15 +457,16 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection, push *model.PushContext,
 		con.proxy.Unlock()
 	}
 
+	configSize := ResourceSize(res)
+	configSizeBytes.With(typeTag.Value(w.TypeUrl)).Record(float64(configSize))
+
 	if err := con.sendDelta(resp); err != nil {
 		recordSendError(w.TypeUrl, con.ConID, err)
 		return err
 	}
 
-	// Some types handle logs inside Generate, skip them here
-	// TODO because we filter out after the fact, SkipLogTypes report wrong info
-	// We should have them return up some metadata that we can transparently log
-	if _, f := SkipLogTypes[w.TypeUrl]; !f {
+	// Some types handle logs inside Generate, skip them here.
+	if !gen.Metadata().LogsDetails {
```

Comment:
This is a little bit confusing

Reply:
What do you suggest? Any idea to make it less confusing?

Reply:
it would be great if the generators could return some "additional metadata to log" or something and still have one place to log IMO